src/execution/physical_plan/plan_comparison_join.cpp

```diff
@@ -281,12 +263,34 @@ unique_ptr<PhysicalOperator> PhysicalPlanGenerator::PlanComparisonJoin(LogicalCo
 			throw NotImplementedException("Unimplemented comparison join");
 		}
 	}
+	return false;
+}
 
+unique_ptr<PhysicalOperator> PhysicalPlanGenerator::PlanComparisonJoin(LogicalComparisonJoin &op) {
+	// now visit the children
+	D_ASSERT(op.children.size() == 2);
+	idx_t lhs_cardinality = op.children[0]->EstimateCardinality(context);
+	idx_t rhs_cardinality = op.children[1]->EstimateCardinality(context);
+	auto left = CreatePlan(*op.children[0]);
+	auto right = CreatePlan(*op.children[1]);
+	left->estimated_cardinality = lhs_cardinality;
+	right->estimated_cardinality = rhs_cardinality;
+	D_ASSERT(left && right);
+
+	if (op.conditions.empty()) {
+		// no conditions: insert a cross product
+		return make_uniq<PhysicalCrossProduct>(op.types, std::move(left), std::move(right), op.estimated_cardinality);
+	}
+
+	size_t has_range = 0;
```

Comment:
Can this be an `idx_t` instead? I know it's the same, but we prefer to use that.